src/components/offers/list/OffersList.js

```diff
@@ -3,17 +3,18 @@ import {ShortOfferDetails} from "./ShortOfferDetails";
 import {useEffect, useState} from "react";
 import {constants} from "../../../utils/constants";
 import {scrollToTop} from '../../../utils/functions';
-import {offersAPI} from "../../../utils/apis/OfferApi";
 
 export const OffersList = (props) => {
 
     const [selectedIdx, setSelectedIdx] = useState(-1)
     const [offers, setOffers] = useState([])
 
+    const getOffers = props.getOffers
+
     useEffect(() => {
-        offersAPI.getAllOffers()
+        getOffers()
             .then(data => setOffers(data || []))
-    }, [])
+    }, [getOffers])
```

Comment:
Czemu w razie niepowodzenia/początkowo tworzymy tablice z funkcją?

Reply:
Dependency Array w useEffect (czyli ta tablica, do której teraz się odnosisz) nie jest chyba tym, co myślisz. Ogólnie ona służy temu, że wtedy useEffect patrzy na zmienne podane w tej tablicy i jeśli one jakkolwiek się zmienią, odpala się jeszcze raz. To nie jest default, jak w useState.

Reply:
Znaczy to inaczej po co tam jest to tablica z funkcją, bo pamiętam że mówiłeś że tu powinna być zazwyczaj pusta tablica.